viewing_party/party.py

```diff
@@ -1,23 +1,352 @@
+
+def get_friends_unique_watched(user_data):
+    '''
+    take one parameter: `user_data`
+    the value of `user_data` will be a dictionary with 
+    a `"watched"` list of movie dictionaries, and a `"friends"`
+    This represents that the user has a list of watched movies 
+    and a list of friends
+    The value of `"friends"` is a list
+    Each item in `"friends"` is a dictionary. 
+    This dictionary has a key `"watched"`, which has a list of movie dictionaries.
+    Each movie dictionary has a `"title"`.
+    Consider the movies that the user has watched, 
+    and consider the movies that their friends have watched. 
+    Determine which movies at least one of the user's friends have watched, 
+    but the user has not watched.
+    Return a list of dictionaries, that represents a list of movies
+
+    '''
+    #friends have watched but user hasn't watched
+    #loop through friend combined list (helper function)
+    #check if friend watched is not in user watched list
+    #and friend_have_watched_user_not list(to prevent duplicate)
+    #if not add to friend_have_watched_user_not list
+
+    friends = user_data["friends"]
+
+    friends_have_watched_user_not = []
+
+    for friend_watch in combined_friends(friends):
+        if (friend_watch not in user_data["watched"] 
+            and friend_watch not in friends_have_watched_user_not):
+
+            friends_have_watched_user_not.append(friend_watch)
+    return friends_have_watched_user_not
+
 
-        
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+    '''
+    takes one parameter: `user_data`
+    `user_data` will have a field `"subscriptions"`. 
+    The value of `"subscriptions"` is a list of strings
+    This represents the names of streaming services that the user has access to
+    Each friend in `"friends"` has a watched list. 
+    Each movie in the watched list has a `"host"`, 
+    which is a string that says what streaming service it's hosted on
+    Determine a list of recommended movies. 
+    A movie should be added to this list if and only if:
+    The user has not watched it
+    At least one of the user's friends has watched
+    The `"host"` of the movie is a service that is in the user's `"subscriptions"`
+    Return the list of recommended movies
+    
+    '''
+    #user_data={"watched":[{}],"subscriptions": ["Netflix", "hulu"]}
+    #friends={"watched":[{“title”: “title A”, “genre”: “horror”, “rating”: 4,"host": "Netflix"}]}
+    #create a list of user subscription and a list of recommended movie
+    #loop through user subscription and append to subscripton list
+    #loop through friend unique watched list and 
+    ##if subscription and not watch, add to recommend list
+    #check if the host exist in user_subscription list
+    #append the movie to the recommended_movies
+    #return recommended movies

```

Comment:
This is lengthy especially since you have docstrings. Your comment should be about one line that explains how the code is working. You can move this to a file that you reference or add them to notes so you understand how the code is working.